sql/parser/sql.y

```diff
@@ -1042,6 +1044,30 @@ show_stmt:
   {
     $$ = &ShowIndex{Table: $4}
   }
+| SHOW GRANTS on_privilege_target_clause for_grantee_clause
+  {
+		$$ = &ShowGrants{Targets: $3, Grantees: $4}
+  }
+
+on_privilege_target_clause:
+  ON privilege_target
+  {
+		$$ = &$2
```

Comment:
So `$2` is a reference to an item in the yacc stack. I'm guessing that something in the parsing is losing this stack item (or clearing it) while you're still holding it. In other words, this isn't kosher. This should probably be:

```
  tmp := $2
  $$ = &tmp
```

or

```
  $$ = &TargetList{}
  *$$ = $2
```

It might be easier if you make `privilege_target` return `*TargetList` instead of a `TargetList`.